cli: extract the SQL connection code to a new package  `clisqlclient` (2/)

knz opened the pull request:
This is part of a family of PRs that aim to encapsulate CLI functionality into finer grained packages.

knz committed:
cli: move SQL client code to `clisqlclient`, new package `clierror`

This commit leaves the code in an intermediate stage, where the unit
tests pass but maybe the linter is unhappy. This will be cleaned up in
the next commit.

Release note: None

knz committed:
clisqlclient: rename some files

Release note: None

knz committed:
clisqlclient: move InitialSQLConnectionError to own file

Release note: None

knz committed:
cli, clisqlclient: only uses public APIs

Release note: None

knz committed:
cli: properly handle conn close errors

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/67457)
<!-- Reviewable:end -->

otan commented on pkg/cli/clierror/cli_error.go:
```diff
@@ -0,0 +1,71 @@
+// Copyright 2021 The Cockroach Authors.
```
nit: should this be called clierror.go, to match the package name?

knz replied:
There are more files coming to it in later PRs, including an `error.go` because there will be an `Error` type. That's the "native" one here.

otan reviewed (commented):
looks good to me, but i did not follow the `rawHTML` too closely and would argue it'd be worth a separate commit

knz committed:
cli: new format 'rawhtml' to auto-gen the settings table

Previously, the cluster setting markdown table was generated with a
`gen` sub-command that was trying to be too clever with the HTML
generator. This was making it complicated to move the table formatting
code to a separate package.

This commit simplifies the code by introducing a new output format
'rawhtml' (raw, as in, it does not escape HTML special characters
inside table cells).

Since this format is only used internally to auto-generate this table,
the help texts leave it undocumented.

Release note: None

knz commented:
> i did not follow the rawHTML too closely and would argue it'd be worth a separate commit

Thanks for the feedback. I have split that change into a separate commit as requested. RFAL.

otan commented on pkg/cli/flags_util.go:
```diff
@@ -323,6 +323,7 @@ const (
 	tableDisplayRecords
 	tableDisplaySQL
 	tableDisplayHTML
+	tableDisplayRawHTML
```
can you add a comment explaining the difference between rawhtml and html?

knz replied:
Done.

knz committed:
clisqlclient: document the table output formats

Release note: None

knz commented:
TFYR!

bors r=otan
